Userland/Applications/Calculator/KeypadValue.cpp

```diff
@@ -107,7 +107,7 @@ KeypadValue::KeypadValue(double d)
     current_pow -= 1;
     while (d != 0) {
         m_value *= 10;
-        m_value += (u64)(d / AK::pow(10.0, (double)current_pow)) % 10;
+        m_value += (u64)(d / AK::pow(10.0, (double)current_pow) + 0.5) % 10;
```

Comment:
maybe prefer a call to `ceil`, due to that essentially being what you want,
just adding a random `.5` does not solve the issue, it only moves it to another example I suppose

Reply:
Hmmm, my initial thought was `ceil` and `floor` would both be unstable to floating point imprecision, that's why I chose rounding to the nearest integer by adding `0.5` and `floor`ing.

For example in the case of `d == 10.0 && currrent_pow = 1.0`  they would give different results depending on
whether the pow function returns `10.00001` or `9.99999`.

Now that I think about it, the "actual" fix for the problem will probably be to not calculate each decimal independently, but instead building the fixed point representation by iteratively adding up the difference to the desired value.

In the meantime I also found this PR, which (I suppose) does exactly that: https://github.com/SerenityOS/serenity/pull/9456

Reply:
than wouldn't a call to `round()` be the right choice?

(standard behavior is truncation btw, although that only matters for unsigned values)

Reply:
hmm, I think in total this really is not the right fix for this. I figured that with rounding to the closest `5/9 == 0.6666`.
I will close this PR and link it on https://github.com/SerenityOS/serenity/pull/9456 for reference.